net/tcp/tcp_sendfile.c

```diff
@@ -495,6 +495,9 @@ ssize_t tcp_sendfile(FAR struct socket *psock, FAR struct file *infile,
    */
 
   net_lock();
+#ifdef CONFIG_NET_TCP_WRITE_BUFFERS
+  conn->sendfile = true;
```

Comment:
should we let sendfile simulate the normal send behavior to avoid adding a new flag?

Reply:
Do you mean to simulate it only in case of "CONFIG_NET_TCP_WRITE_BUFFERS=y" mode?

Reply:
I have the following plan concerning TCP rework:
1. Fix already found issues in tcp_sendfile. One of the fix is the current PR #5239.
2. Also I have almost prepared the next patch that fixes another issue in tcp_sendfile (+ code optimization), and at the same time the TCP payload throughput of tcp_sendfile is increased from 234 Mbits/s up to about 432 Mbit/s (sim:tcpblaster config on my Linux setup).
3. Fix the rest of already found issues in tcp_sendfile.
4. Try to modify tcp_send_unbuffered.c to achieve similar TCP payload throughput as already achieved for tcp_sendfile.
5. Try to modify tcp_send_buffered.c to achieve similar TCP payload throughput as already achieved for tcp_sendfile.
6. If TCP payload throughput for tcp_send_buffered.c is achieved at least as tcp_sendfile already shows, then I could try to simulate tcp_sendfile over tcp_send_buffered.c (CONFIG_NET_TCP_WRITE_BUFFERS=y) and check if the throughput is not reduced because of the simulation.

Right now in case of sim:tcpblaster config the TCP throughput is only 36.7 Kbit/s. If I try to simulate tcp_sendfile over tcp_send_buffered.c right now, the throughput of tcp_sendfile will not be higher 36.7 Kbit/s for sure.

Thus I'm suggesting to go step by step. I'm going to prepare the patches as I outlined.

Reply:
So, the different code path from this PR and #5138 will merge again in the final?

Reply:
Concerning PR #5138 I think there is another situation. I expect that controlling sndseq can be done the same way (unified) without dependence on CONFIG_NET_TCP_WRITE_BUFFERS option. So far I have not touched the buffered mode concerning sndseq control as it seems work. However, I'm going to consider unification of how sndseq is controlled as well.

Reply:
> 2\. Also I have almost prepared the next patch that fixes another issue in tcp_sendfile (+ code optimization), and at the same time the TCP payload throughput of tcp_sendfile is increased from 234 Mbits/s up to about 432 Mbit/s (sim:tcpblaster config on my Linux setup).

I created PR #5242.

Reply:
Thanks for taking the effort to unify the code path to simplify the maintain in the future.